content/modules/rest/pages/entities-api.adoc

```diff
@@ -0,0 +1,724 @@
+</queries>
+----
+
+A query needs to have a unique `name` value as well as an `entity` reference. The combination of `name` and `entity` needs to be unique. Also, a `fetchPlan` needs to be referenced to indicate which entity attributes are returned.
+
+In the `<jpql>` tag the actual query is configured. Parameters are referenced via their name prefixed with a colon like `:customerName`.
+The parameters itself need to be listed within the `params` tag defining their name and Java type.
+
+The Entity Query endpoint can be invoked by the `GET` or `POST` HTTP method. In case of the `GET`, the parameters are appended
+as URL query parameters.
+
+[source, http request]
+.Query API GET Request
+----
+GET http://localhost:8080/rest
+        /queries
+        /rstex11_Order
+        /ordersByDate
+        ?orderDate=2020-02-02
+----
+
+
+[IMPORTANT]
+.URI Encoding
+====
+Similar to the Entity Search Endpoint the URL should only contain ASCII characters. This means the values of the parameters need to be URL encoded, as those values normally represent direct user input and therefore it cannot be ensured that non-ascii characters are used.
+====
+
+In the case of using `POST`, the query parameters are transferred in the JSON body containing each parameter as a key.
+
+[source, http request]
+.Query API POST Request
+----
+POST http://localhost:8080/rest/queries/rstex11_Order/ordersByCustomerName
+
+{
+  "customerName": "Shelby Robinson"
+}
+----
+
+==== Collection Parameters
+
+It is also possible to define a parameter as a collection type. In this case the query definition should contain a `[]` indicator after the Java type.
+
+[source,xml]
+.rest-queries.xml
+----
+<?xml version="1.0"?>
+<queries xmlns="http://jmix.io/schema/rest/queries">
+    <query name="ordersByIds" entity="rstex11_Order" fetchPlan="order-with-details">
+        <jpql><![CDATA[select e from rstex11_Order e where e.id in :ids]]></jpql>
+        <params>
+            <param name="ids" type="java.util.UUID[]"/> // <1>
+        </params>
+    </query>
+</queries>
+----
+<1> the `ids` parameter is marked as collection of `UUID` type
+
+When this parameter is used in a query, the corresponding  IDs have to be provided as a JSON array.
+
+[source, http request]
+.Query API Collection Parameters Request
+----
+POST http://localhost:8080/rest/queries/rstex11_Order/ordersByIds
+
+{
+  "ids": [
+    "41aae331-b46b-85ee-b0bc-2de8cbf1ab86",
+    "21021f78-edac-224b-e6f8-6e71e02a0f0d"
+  ]
+}
+----
+
+== Create Entities
+
+The Entities API allows creating entities through the Create Entity endpoint. An entity can be created via performing a `POST` request against the endpoint: `/entities/:entityName`.
+
+The request body contains a JSON object with the attributes of the entity.
+
+NOTE: Attributes from xref:data-model:entities.adoc#traits[entity traits] like `id` or `createdBy` should not be part of the request. Those attributes are automatically added by Jmix while storing the entity.
```

Comment:
Actually, if the entity's id attribute has no `JmixGeneratedValue` or `GeneratedValue` annotations, the client has to provide an id value.

Reply:
ok, I can extend this. But how is this for the other entity trait attributes? Because for those there is no need to use a `@GeneratedValue` annotation, right? How would you suggest to write it  then?

Reply:
Perhaps it depends on the trait.
For Audit and Soft Delete traits it's simple - the client should not send them.
The others are complicated:
- HasUUID trait includes id if it's of UUID type. Let's not mention it at all.
- Version should not be send on create but should be send on update otherwise optimictic lock won't work. Worth mentioning.